homeassistant/components/upnp/sensor.py

```diff
@@ -112,6 +113,9 @@ def device_info(self):
             'identifiers': {
                 (DOMAIN_UPNP, self.unique_id)
             },
+            'connections': {
+                (dr.CONNECTION_UPNP, self._device.udn)
```

Comment:
This will merge them with the hub, and in that case you should remove `via_hub` ?

Reply:
That is not intended. How do I properly make the sensors a 'child' of the main device, if possible? @balloob 

If I add some logging/debug statements to `DeviceRegistry.async_get_or_create()` and supply `via_hub` from the sensors, it does find the hub device. The result, however, is as seen in the screenshots above.

Reply:
There are two different concepts.

- via_hub indicates that this device communicates via another device to Home Assistant. Example is a Hue light that uses the Hue hub to connect to Home Assistant
- identifiers/connections are part of a device. The device will be merged with any other device that has an overlapping identifier/connection.

Reply:
In this case it is a device (the IGD/router itself) with traffic counters (the sensors.) Does home assistant have any guidelines how to handle this? It seems to me either it should be considered part of a device, or not at all. Certainly not as a hub.

Reply:
We just updated the docs last week on this: https://developers.home-assistant.io/docs/en/next/device_registry_index.html#what-is-a-device

Although, I guess this doesn't really fit in that format 🙄 

I think that these sensors should, if possible, use the mac address as an identifier for the device. They should not use same identifier/connection as router, as it's not about that? Although, one might argue it is connected to the router, as it keeps track of how much data is used at that device by some other device?

Reply:
I think they should use the UDN (unique device name.) A single (physical) device can provide multiple UPnP devices.

The traffic counters are part of the device. They report traffic counters for this (UPnP) device. I think they should be merged (connected) then.